callflow/operations/unify.py

```diff
@@ -65,23 +75,23 @@ def compute(self):
             # ------------------------------------------------------------------
             # unify the dataframe
             # remap the modules in this supergraph to the one in ensemble graph
-            sg_modules_list = np.array(list(sg.module2idx.keys()))
-            sg_callsites_list = np.array(list(sg.callsite2idx.keys()))
+            sg_modules_list = self._remove_none(list(sg.module2idx.keys()))
+            sg_callsites_list = self._remove_none(list(sg.callsite2idx.keys()))
             _mod_map = create_reindex_map(sg_modules_list, self.eg_modules_list)
             _cs_map = create_reindex_map(sg_callsites_list, self.eg_callsites_list)
-
+            
             if 1:  # edit directly in the supergraph
                 sg.df_add_column("dataset", apply_value=sg.name)
                 sg.df_add_column(
                     "module",
                     update=True,
-                    apply_func=lambda _: self.eg_modules_list[_mod_map[_]],
+                    apply_func=lambda _: self.eg_modules_list[_mod_map[_]] if _ is not -1 else -1,
```

Comment:
the `if` statement is there to avoid overriding the rows for which `module == -1`. We don't have to replace them with the new mapping.